dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/DatasetsValidators.java

```diff
@@ -0,0 +1,137 @@
+package edu.harvard.iq.dataverse.api;
+
+import com.amazonaws.services.pi.model.InvalidArgumentException;
+import edu.harvard.iq.dataverse.api.dto.FileTermsOfUseDTO;
+import edu.harvard.iq.dataverse.common.BundleUtil;
+import edu.harvard.iq.dataverse.license.TermsOfUseFactory;
+import edu.harvard.iq.dataverse.persistence.datafile.license.FileTermsOfUse;
+import edu.harvard.iq.dataverse.persistence.datafile.license.LicenseDAO;
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean;
+import org.apache.commons.lang.StringUtils;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import java.security.InvalidParameterException;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+
+@Stateless
+public class DatasetsValidators {
+    private LicenseDAO licenseDAO;
+    private TermsOfUseFactory termsOfUseFactory;
+    private SettingsServiceBean settingsService;
+
+    // -------------------- CONSTRUCTORS --------------------
+    @Deprecated
+    public DatasetsValidators() {
+    }
+
+    @Inject
+    public DatasetsValidators(LicenseDAO licenseDAO, TermsOfUseFactory termsOfUseFactory, SettingsServiceBean settingsService) {
+        this.licenseDAO = licenseDAO;
+        this.termsOfUseFactory = termsOfUseFactory;
+        this.settingsService = settingsService;
+    }
+
+    // -------------------- LOGIC --------------------
+    public void validateFileTermsOfUseDTO(FileTermsOfUseDTO fileTermsOfUseDTO) throws InvalidParameterException {
+        if(StringUtils.isEmpty(fileTermsOfUseDTO.getTermsType())) {
+            throw new InvalidParameterException(BundleUtil.getStringFromBundle("datasets.api.add.termsOfUseAndAccess.missingTermsOfUse"));
```

Comment:
We shouldn't use bundles in exceptions, they are internal errors. You can use some exception resolver that will based on the type of exception return certain message. You can also return the list of strings which will contain the errors and then you can parse them in API Class.

Reply:
We have agreed to store these messages in bundles. There is nothing wrong in using exception error message. I'm validating object, when validation fails an exception is being thrown and it only happens that I can use its message for what I want and don't have to parse it anyhow.

IMO Adding different exception for each message or parsing error strings would be both unclean and an overkill.

Reply:
Well we had this talk before with Krzysiek and he also agreed with that. The problem is not with storing messages in bundles but using exceptions to pass them. Exceptions are internal errors and they shouldn't show errors in different languages, how are we going to read them in logs if they could be in spanish for example? I agree that handling different exceptions is overkill that's why I think returning List of errors or String is sufficient.

Reply:
But the thing is - we don't want to translate them (ever). We only use bundles to store them in one place and manage them there. There shouldn't be versions of 'api' bundles anywhere else but default english bundle.

Reply:
Ahh ok, then it is a different story, I'm ok with that then.